src/hooks/usePatientPrograms.ts

```diff
@@ -0,0 +1,32 @@
+import { openmrsFetch, restBaseUrl } from '@openmrs/esm-framework';
+import { useEffect, useState } from 'react';
+import { type FormSchema, type PatientProgram } from '../types';
+const customRepresentation = `custom:(uuid,display,program:(uuid,name,allWorkflows),dateEnrolled,dateCompleted,location:(uuid,display),states:(startDate,endDate,state:(uuid,name,retired,concept:(uuid),programWorkflow:(uuid)))`;
+
+export const usePatientPrograms = (patientUuid: string, formJson: FormSchema) => {
+  const [patientPrograms, setPatientPrograms] = useState<Array<PatientProgram>>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState(null);
```

Comment:
Why not use SWR loading and error states instate of useEffect

Reply:
SWR loads stale data, that's why the `useEncounter`  hook was refactored to use openMrsFetch, [see](https://openmrs.atlassian.net/browse/O3-3164).

Reply:
makes sense